app/components/tables/holidays_table/HolidaysTable.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+    useReactTable,
+    flexRender,
+    getCoreRowModel,
+    getPaginationRowModel,
+    getFilteredRowModel,
+    getSortedRowModel,
+} from "@tanstack/react-table";
+import useAxiosPromise from "../../../hooks/useAxiosPromise";
+import React, { useMemo, useState, useEffect } from "react";
+import HolidaysTableHeaders from "@/components/tables/holidays_table/table_element/HolidaysTableHeaders";
+import { Holiday } from "../../../types/Holiday";
+interface HolidaysTableProps {
+    data: Holiday[];
+}
```

Comment:
If the Holiday type is only being used in this HolidaysTable, it should be defined here instead of in the /types folder. Then we can go ahead and delete the /types folder.